Library/Source/NativeEngine.h

```diff
@@ -22,35 +22,28 @@ namespace babylon
     {
     public:
         RecycleSet(T firstId)
-            : m_nextId{ firstId }
+            : m_firstId{ firstId }
+            , m_nextId{ firstId }
         {}
 
         RecycleSet() : RecycleSet({ 0 })
         {}
 
         T Get()
         {
-            if (m_queue.empty())
-            {
-                return m_nextId++;
-            }
-            else
-            {
-                T next = m_queue.front();
-                m_queue.pop();
-                return next;
-            }
+            m_nextId ++;
+            assert(m_nextId < bgfx::getCaps()->limits.maxViews);
+            return m_nextId;
         }
 
-        void Recycle(T id)
+        void Clear()
         {
-            assert(id < m_nextId);
-            m_queue.push(id);
+            m_nextId = m_firstId;
```

Comment:
With this change, I don't think the `RecycleSet` is adding value anymore. The reason it was created was to abstract away the maintenance of state; but if we're just throwing out all the state every frame anyway, the `FrameBufferManager` can probably just do that directly; it can then have asserts that check ViewIDs and thinks like that without creating any errant dependencies. In short, if this is the behavior we're using for ViewIDs now, I think the `RecycleSet` should probably be deleted.

Reply:
I removed RecycleSet and changed some method names. It's a bit more clear now.